vector/src/main/java/im/vector/fragments/roomwidgets/RoomWidgetPermissionViewModel.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.fragments.roomwidgets
+
+import com.airbnb.mvrx.BaseMvRxViewModel
+import com.airbnb.mvrx.MvRxState
+import com.airbnb.mvrx.MvRxViewModelFactory
+import com.airbnb.mvrx.ViewModelContext
+import im.vector.Matrix
+import im.vector.R
+import im.vector.widgets.Widget
+import org.matrix.androidsdk.MXSession
+import org.matrix.androidsdk.core.Log
+import org.matrix.androidsdk.core.callback.ApiCallback
+import org.matrix.androidsdk.core.model.MatrixError
+import java.net.URL
+
+data class RoomWidgetPermissionViewState(
+        val authorId: String = "",
+        val authorAvatarUrl: String? = null,
+        val authorName: String? = null,
+        val widgetDomain: String? = null,
+        //List of @StringRes
+        val permissionsList: List<Int>? = null
+) : MvRxState
+
+class RoomWidgetPermissionViewModel(val session: MXSession, val widget: Widget, initialState: RoomWidgetPermissionViewState)
+    : BaseMvRxViewModel<RoomWidgetPermissionViewState>(initialState, false) {
+
+    init {
+        val room = session.dataHandler.getRoom(widget.roomId)
+        val creator = room.getMember(widget.widgetEvent.sender)
+
+        var domain: String?
+        try {
+            domain = URL(widget.url).host
+        } catch (e: Throwable) {
+            domain = null
+        }
+
+        //TODO check from widget urls the perms that should be shown?
+        //For now put all
+        val infoShared = listOf<Int>(
+                R.string.room_widget_permission_display_name,
+                R.string.room_widget_permission_avatar_url,
+                R.string.room_widget_permission_user_id,
+                R.string.room_widget_permission_theme,
+                R.string.room_widget_permission_widget_id,
+                R.string.room_widget_permission_room_id
+        )
+        setState {
```

Comment:
This should be done in the initialState and not here

Reply:
moved

Reply:
thx